deployment-manager/src/main/java/com/example/deploymentmanager/DeploymentManagerDemo.java

```diff
@@ -0,0 +1,124 @@
+ * limitations under the License.
+ */
+
+package com.example.deploymentmanager;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.googleapis.auth.oauth2.GoogleAuthorizationCodeFlow;
+import com.google.api.client.googleapis.auth.oauth2.GoogleClientSecrets;
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpRequestInitializer;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.deploymentmanager.DeploymentManager;
+import com.google.api.services.deploymentmanager.model.*;
+
+import java.io.*;
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class DeploymentManagerDemo {
+
+  private static String CONFIG_YAML = "yaml/deployment-manager-config.yaml";
+  private static String DEPLOYMENT = "my-deployment-1";
+
+  public static void main(String args[]) throws GeneralSecurityException, IOException {
+
+    String GOOGLE_PROJECT_ID = System.getenv("GOOGLE_PROJECT_ID");
+    if (GOOGLE_PROJECT_ID == null || GOOGLE_PROJECT_ID.compareTo("") == 0) {‰
+      System.out.println("GOOGLE_PROJECT_ID is not set");
+      System.exit(-1);
+    }
+
+    // best practice is to use application default credential. If inside a GCE
+    // instance it is set automatically.
+    
+    GoogleCredential credential = GoogleCredential.getApplicationDefault();
+
+    // to load from somewhere else use this
+    // GoogleCredential credential = GoogleCredential.fromStream(new
+    // FileInputStream(path_to_key));
+
+    HttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
+    JsonFactory jsonFactory = JacksonFactory.getDefaultInstance();
+
+    DeploymentManager dm2 = new DeploymentManager.Builder(httpTransport, jsonFactory, credential)
+        .setApplicationName("New_Application").build();
+
+    Deployment deploy = new Deployment();
+    // Optional we want some labels: Map of labels; provided by the client when the
+    // resource is created or updated.
+    List<DeploymentLabelEntry> labels = new ArrayList<DeploymentLabelEntry>();
+
+    labels.add(new DeploymentLabelEntry().setKey("test-label-1").setValue("good-fella"));
+
+    deploy.setLabels(labels);
+
+    // load the config file contents and setup target conrfiguration
+    ConfigFile config = new ConfigFile().setContent(readYaml());
+
+    TargetConfiguration target = new TargetConfiguration().setConfig(config);
+
+    deploy.setTarget(target);
+    deploy.setName(DEPLOYMENT);
+
+    List<Deployment> deployments = dm2.deployments().list(GOOGLE_PROJECT_ID).execute().getDeployments();
```

Comment:
Please extract `dm2.deployments().list(GOOGLE_PROJECT_ID).execute()` in to a separate statement, since this is an API call. It likely needs it's own try/catch function, as well as some validation the response was received correctly before calling `getDeployements()`.